components/CheckDocumentation/vacuum/XminHorizon.tsx

```diff
@@ -0,0 +1,342 @@
+      {heldBackInfo && (
+        <p>
+          A replication slot is holding back the xmin horizon at{" "}
+          <code>{heldBackInfo["xmin"]}</code>, specifically with system
+          catalogs.
+        </p>
+      )}
+      <p>
+        You can check the replication status on the{" "}
+        <Link to={serverReplicationUrl}>Replication</Link> page. You may also
+        want to check logs on both the publisher and the subscriber for any
+        error messages regarding logical replication, such as schema
+        differences.
+      </p>
+      <p>
+        If the replication slot is no longer used, remove it it by running the
+        following command:
+      </p>
+      <CodeBlock>
+        <SQL sql={`SELECT pg_drop_replication_slot('<slot_name>');`} />
+      </CodeBlock>
+    </li>
+  );
+};
+
+const GuidanceByStandby: React.FunctionComponent<{
+  inApp: boolean;
+  heldBackInfo: number | null;
+}> = ({ inApp, heldBackInfo }) => {
+  if (inApp && !heldBackInfo) {
+    return null;
+  }
+
+  const CodeBlock = useCodeBlock();
+  return (
+    <li>
+      <h5>Long-running queries on standbys</h5>
+      <p>
+        When <code>hot_standby_feedback</code> is on with physical streaming
+        replication, queries on standbys will hold back the xmin horizon just as
+        if they were running on the primary.
+      </p>
+      <h6>
+        <b>Solution</b>
+      </h6>
+      {heldBackInfo && (
+        <p>
+          A long running query on a standby is holding back the xmin horizon at{" "}
+          <code>{heldBackInfo["xmin"]}</code>.
+        </p>
+      )}
+      <p>
+        You can find the <code>xmin</code> of all standby servers by running the
+        following command:
+      </p>
+      <CodeBlock>
+        <SQL
+          sql={`SELECT application_name, client_addr, backend_xmin
+                  FROM pg_stat_replication
+                  ORDER BY age(backend_xmin) DESC;`}
+        />
+      </CodeBlock>
+      <p>
+        Once the standby is identified, you can find the query holding back the
+        xmin horizon and its connection's pid in that standby by running the
+        following command:
+      </p>
+      <CodeBlock>
+        <SQL
+          sql={`SELECT pid, datname, usename, state, backend_xmin, backend_xid
+                  FROM pg_stat_activity
+                  WHERE backend_xmin IS NOT NULL OR backend_xid IS NOT NULL
+                  ORDER BY greatest(age(backend_xmin), age(backend_xid)) DESC;`}
+        />
+      </CodeBlock>
+    </li>
+  );
+};
+
+const GuidanceByPreparedXact: React.FunctionComponent<{
+  inApp: boolean;
+  heldBackInfo: number | null;
+}> = ({ inApp, heldBackInfo }) => {
+  if (inApp && !heldBackInfo) {
+    return null;
+  }
+
+  const CodeBlock = useCodeBlock();
+  return (
+    <li>
+      <h5>Abandoned prepared transactions</h5>
+      <p>
+        A transaction prepared for a two-phase commit will prevent cleanup until
+        it is either committed or rolled back.
```

Comment:
Since prepared transactions are not very common, and it's possible the DBA who is alerted about this issue is not familiar with them (even though the application programmer who caused the problem must have been), should we link to, e.g., https://www.postgresql.org/docs/current/sql-prepare-transaction.html somewhere here?

Reply:
I was like "I'm pretty positive I got this feedback before" and realized that it was with my blog post 🙃 I'll add a link 👍